src/runner.py

```diff
@@ -369,6 +369,15 @@ def _install_binary(self, binary: Path) -> None:
         logger.info("Installing Python 3 pip...")
         self._execute(["/usr/bin/apt", "install", "-yq", "python3-pip"])
 
+        logger.info("Installing shellcheck")
+        self._execute(["/usr/bin/apt", "install", "-yq", "shellcheck"])
```

Comment:
Perhaps it is worth adding a wrapping function around `self._execute` to install `apt` and `snap` packages that do the logging etc so that we can simplify this a bit? Or maybe that function even takes a list of apt packages and then installs them?